src/context/AuthContext.tsx

```diff
@@ -0,0 +1,87 @@
+'use client';
+
+import React, {
+  createContext,
+  useState,
+  useContext,
+  useEffect,
+  ReactNode,
+} from 'react';
+
+import { getUser } from '../client';
+import { User } from '../types';
+
+type AuthContextProps = {
+  user: User | null;
+  login: (token: string) => void;
+  logout: () => void;
+};
+
+export const AuthContext = createContext<AuthContextProps | undefined>(
+  undefined,
+);
+
+type AuthProviderProps = {
+  children: ReactNode;
+};
+
+export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const getCookie = (name: string): string | null => {
+    if (typeof document === 'undefined') {
+      return null;
+    }
+
+    const value = `; ${document.cookie}`;
+    const parts = value.split(`; ${name}=`);
+
+    return parts.length === 2 ? parts?.pop()?.split(';')?.[0] || null : null;
+  };
+
+  const setCookie = (name: string, value: string, days = 7): void => {
+    if (typeof document === 'undefined') {
+      return;
+    }
+    const date = new Date();
+    date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000);
+    const expires = `; expires=${date.toUTCString()}`;
+    document.cookie = `${name}=${value}${expires}; path=/; Secure; HttpOnly`;
+  };
```

Comment:
That's a lot of implementation! :D Use nookies so we can get rid of this -> https://github.com/maticzav/nookies, hopefully, it will remove some of that code.

Reply:
Oh gosh, I should have asked you first if I could use something for that.  :D